common/src/main/kotlin/org/neo4j/connectors/kafka/data/ChangeEventExtensions.kt

```diff
@@ -396,44 +459,68 @@ internal fun Struct.toRelationshipEvent(): RelationshipEvent =
           after)
     }
 
-@Suppress("UNCHECKED_CAST")
+@Suppress("UNCHECKED_CAST", "IMPLICIT_CAST_TO_ANY")
 internal fun Struct.toNodeState(): Pair<NodeState?, NodeState?> =
     Pair(
         getStruct("before")?.let {
           val labels = it.getArray<String>("labels")
-          val properties = it.getMap<String, Any?>("properties")
+          val propertiesField = it.schema().field("properties")
+          val properties =
+              when (propertiesField.schema().type()) {
+                Schema.Type.MAP -> it.getMap<String, Any?>("properties")
```

Comment:
why not use `payloadMode` here instead of checking the `properties` schema type? would that work?

Reply:
Yes, sure it would work but in this case we should introduce the same configuration for sink connector as well. These methods are being used for the conversions from Struct (kafka connect object) to CdcEvent (that we manage on our side) and I think it's not necessary to introduce this configuration for sink connector to do conversions since we can determine data structure from the schema.